Source/Core/AudioCommon/Mixer.cpp

```diff
@@ -119,12 +129,63 @@ unsigned int CMixer::Mix(short* samples, unsigned int num_samples, bool consider
 
   memset(samples, 0, num_samples * 2 * sizeof(short));
 
-  m_dma_mixer.Mix(samples, num_samples, consider_framelimit);
-  m_streaming_mixer.Mix(samples, num_samples, consider_framelimit);
-  m_wiimote_speaker_mixer.Mix(samples, num_samples, consider_framelimit);
+  unsigned int actual_samples = m_dma_mixer.Mix(samples, num_samples, false);
+  m_streaming_mixer.Mix(samples, num_samples, false);
+  m_wiimote_speaker_mixer.Mix(samples, num_samples, false);
+
+  StretchAudio(samples, actual_samples, num_samples);
```

Comment:
Are you sure you're allowed to readback the values in samples? I mean it could be write-combined memory shared with the audio hardware.
Maybe just use a temporary buffer here.